### PR TITLE
feat(components/text-editor): add hint text to text editor component

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#2837811](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2837811)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2257/storybook/)

Component Storybooks:

- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2257/storybooks/text-editor-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2257/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2257/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2257/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2257/playground/)

File: libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.html [hunk 1]
```diff
@@ -63,6 +64,11 @@
   >
   </iframe>
 </div>
+<div skyId #hintTextEl="skyId">
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.html]: Same advice as this: https://github.com/blackbaud/skyux/pull/2252/files#r1581371417

File: libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.scss [hunk 2]
```diff
@@ -69,6 +69,10 @@
   }
 }
 
+.sky-text-editor-hint-text {
+  margin-top: var(--sky-margin-stacked-xs);
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.scss]: I suggest removing this style and using `sky-margin-stacked-xs` directly on the element above the hint text element when hint text is specified.

Comment (Blackbaud-CoreyArcher) [reply libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.scss]: removed

File: libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.html (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.html]: updated

**Review by Blackbaud-PaulCrowder — DISMISSED** [review]

**Review by Blackbaud-PaulCrowder — APPROVED** [review]

**Pull request merged**